core/src/main/java/tc/oc/pgm/gamerules/GameRulesMatchModule.java

```diff
@@ -21,9 +20,18 @@ public void load() {
     for (Map.Entry<String, String> gameRule : this.gameRules.entrySet()) {
       this.match.getWorld().setGameRuleValue(gameRule.getKey(), gameRule.getValue());
     }
+
+    /* Gets gamerule values from level.dat after being set
+     Maps need doDaylightCycle set in XML for it to be turned on, so doDayLightCycle from level.dat is not saved
+    */
+    for (String gameRule : this.match.getWorld().getGameRules()) {
+      if (!gameRule.equals("doDaylightCycle")) {
+        gameRules.put(gameRule, this.match.getWorld().getGameRuleValue(gameRule));
+      }
+    }
   }
```

Comment:
The order of these 2 loops needs to be inverted.

First you should set the gamerules from the world (as fallback) (match.getWorld.getGameRules)
Then override them with the gamerules defined in the XML (this.gamerules)
Lastly, force the doDaylightCycle gamerule to if the XML has timelock on or off. (make sure you define GameRulesModule having a dependency on WorldTimeModule)

There's no need to do the check for doDaylightCycle on each iteration